solc/jsonCompiler.cpp

```diff
@@ -203,51 +203,65 @@ string compile(StringMap const& _sources, bool _optimize, CStyleReadFileCallback
 
 	if (success)
 	{
-		output["contracts"] = Json::Value(Json::objectValue);
-		for (string const& contractName: compiler.contractNames())
+		try
 		{
-			Json::Value contractData(Json::objectValue);
-			contractData["interface"] = compiler.interface(contractName);
-			contractData["bytecode"] = compiler.object(contractName).toHex();
-			contractData["runtimeBytecode"] = compiler.runtimeObject(contractName).toHex();
-			contractData["opcodes"] = solidity::disassemble(compiler.object(contractName).bytecode);
-			contractData["functionHashes"] = functionHashes(compiler.contractDefinition(contractName));
-			contractData["gasEstimates"] = estimateGas(compiler, contractName);
-			auto sourceMap = compiler.sourceMapping(contractName);
-			contractData["srcmap"] = sourceMap ? *sourceMap : "";
-			auto runtimeSourceMap = compiler.runtimeSourceMapping(contractName);
-			contractData["srcmapRuntime"] = runtimeSourceMap ? *runtimeSourceMap : "";
-			ostringstream unused;
-			contractData["assembly"] = compiler.streamAssembly(unused, contractName, _sources, true);
-			output["contracts"][contractName] = contractData;
-		}
+			output["contracts"] = Json::Value(Json::objectValue);
+			for (string const& contractName: compiler.contractNames())
+			{
+				Json::Value contractData(Json::objectValue);
+				contractData["interface"] = compiler.interface(contractName);
+				contractData["bytecode"] = compiler.object(contractName).toHex();
+				contractData["runtimeBytecode"] = compiler.runtimeObject(contractName).toHex();
+				contractData["opcodes"] = solidity::disassemble(compiler.object(contractName).bytecode);
+				contractData["functionHashes"] = functionHashes(compiler.contractDefinition(contractName));
+				contractData["gasEstimates"] = estimateGas(compiler, contractName);
+				auto sourceMap = compiler.sourceMapping(contractName);
+				contractData["srcmap"] = sourceMap ? *sourceMap : "";
+				auto runtimeSourceMap = compiler.runtimeSourceMapping(contractName);
+				contractData["srcmapRuntime"] = runtimeSourceMap ? *runtimeSourceMap : "";
+				ostringstream unused;
+				contractData["assembly"] = compiler.streamAssembly(unused, contractName, _sources, true);
+				output["contracts"][contractName] = contractData;
+			}
+
+			// Do not taint the internal error list
+			ErrorList formalErrors;
+			if (compiler.prepareFormalAnalysis(&formalErrors))
+				output["formal"]["why3"] = compiler.formalTranslation();
```

Comment:
I'm fine to merge whichever ways (unified or separate).

Is there a standard test of regression tests that should be run before every merge?